src/validators/mod.rs

```diff
@@ -740,3 +720,71 @@ pub trait Validator: Send + Sync + Debug {
     /// it is used by `UnionValidator` to calculate strictness
     fn complete(&self) -> PyResult<()>;
 }
+
+pub struct OuterValidator {
+    pub inner: CombinedValidator,
+}
+
+impl Validator for OuterValidator {
+    fn validate<'data>(
+        &self,
+        py: Python<'data>,
+        input: &'data impl Input<'data>,
+        state: &mut ValidationState,
+    ) -> ValResult<'data, PyObject> {
+        let mut token = state.descend();
+        self.inner.validate(py, input, token.get_state()?)
+    }
+
+    fn validate_assignment<'data>(
+        &self,
+        py: Python<'data>,
+        obj: &'data PyAny,
+        field_name: &'data str,
+        field_value: &'data PyAny,
+        state: &mut ValidationState,
+    ) -> ValResult<'data, PyObject> {
+        let mut token = state.descend();
+        self.inner
+            .validate_assignment(py, obj, field_name, field_value, token.get_state()?)
+    }
+
+    fn default_value<'data>(
+        &self,
+        _py: Python<'data>,
+        _outer_loc: Option<impl Into<LocItem>>,
+        _state: &mut ValidationState,
+    ) -> ValResult<'data, Option<PyObject>> {
+        self.inner.default_value(_py, _outer_loc, _state)
+    }
+
+    fn different_strict_behavior(&self, ultra_strict: bool) -> bool {
+        self.inner.different_strict_behavior(ultra_strict)
+    }
+
+    fn get_name(&self) -> &str {
+        self.inner.get_name()
+    }
+
+    fn complete(&self) -> PyResult<()> {
+        self.inner.complete()
+    }
+}
+
+impl OuterValidator {
+    pub fn new(inner: CombinedValidator) -> Self {
+        Self { inner }
+    }
+}
+
+impl PyGcTraverse for OuterValidator {
+    fn py_gc_traverse(&self, visit: &PyVisit<'_>) -> Result<(), PyTraverseError> {
+        self.inner.py_gc_traverse(visit)
+    }
+}
+
+impl Debug for OuterValidator {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        self.inner.fmt(f)
+    }
+}
```

Comment:
Torn on this. We could just derive it but then `OuterValidator` shows up in some python reprs which seems like an unnecessary implementation detail. We'd only have to update 1 test though.

Reply:
I think this is probably good.